### PR TITLE
feat(types): add initial Blob implementation

### PR DESCRIPTION
**Pull request by oblique**
Depends on #1

### CODE REVIEW

File: types/src/blob.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        let blob_json = BlobJson::deserialize(deserializer)?;
+
+        let blob = Blob {
+            blob: tmtypes::Blob {
+                namespace_id: blob_json.namespace[1..].to_vec(),
+                namespace_version: blob_json.namespace[0] as u32,
+                data: blob_json.data.clone(),
+                share_version: blob_json.share_version,
+            },
+            commitment: blob_json.commitment.clone(),
```
Comment (zvolin) [inline types/src/blob.rs:52]:
```suggestion
                data: blob_json.data,
                share_version: blob_json.share_version,
            },
            commitment: blob_json.commitment,
```
Clones shouldn't be needed

File: types/Cargo.toml [hunk 2]
```diff
@@ -4,4 +4,7 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
+base64 = "0.21.2"
 celestia-proto = { workspace = true }
+serde = { version = "1.0.164", features = ["derive"] }
+serde_json = "1.0.97"
```
[the remark this answers is not in the document]
Comment (oblique) [reply types/Cargo.toml]: i removed it

File: types/src/blob.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
+use celestia_proto::tendermint::types as tmtypes;
+use serde::{Deserialize, Deserializer, Serialize};
+
+pub struct Blob {
+    blob: tmtypes::Blob,
+    commitment: Vec<u8>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct BlobJson {
```
[the remark this answers is not in the document]
Comment (oblique) [reply types/src/blob.rs:13]: Added a [reference](https://github.com/celestiaorg/celestia-node/blob/6654cdf4994dbd381efd0d6a29688c731177c855/blob/blob.go#L134-L164) to the Go implementation

**Review by zvolin — APPROVED** [review]
LGTM

**Pull request merged**